src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -60,9 +60,11 @@ class RelayDefaultNetworkLayer {
       } else {
         request.resolve({response: payload.data});
       }
-    }).catch(
-      error => request.reject(error)
-    );
+      return request;
+    }).catch(error => {
+      request.reject(error);
+      return request;
+    });
```

Comment:
Right.. this would work..  so just [example](https://facebook.github.io/relay/docs/api-reference-relay.html#injectnetworklayer-static-method-example) misled me to using promise returned from `DefaultNetworkLayer.sendMutation`.